### PR TITLE
Added more tests to verify null equality join support

### PR DESCRIPTION
**Pull request by khurram-faraaz**
Hello Abhishek,

I have added some more new tests to verify support for null equality joins.
Please review on this branch and merge. I will share the link to pre-commit run.

Thanks,
Khurram

### CODE REVIEW

File: framework/resources/Datasources/join/crt_tbls_partition_by_r.ddl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+DROP TABLE IF EXISTS r_tblprtnby_intcl;
+CREATE TABLE r_tblprtnby_intcl PARTITION BY( col_int ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_r`;
+--DROP TABLE IF EXISTS r_tblprtnby_chrcl;
+--CREATE TABLE r_tblprtnby_chrcl PARTITION BY( col_chr ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_r`;
+--DROP TABLE IF EXISTS r_tblprtnby_vrchrcl1;
+--CREATE TABLE r_tblprtnby_vrchrcl1 PARTITION BY( col_vrchr1 ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_r`;
+--DROP TABLE IF EXISTS r_tblprtnby_vrchrcl2;
+--CREATE TABLE r_tblprtnby_vrchrcl2 PARTITION BY( col_vrchr2 ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_r`;
```
Comment (Agirish) [inline framework/resources/Datasources/join/crt_tbls_partition_by_r.ddl]: Can we remove all commented out lines? Any reason you have them here?

File: framework/resources/Datasources/join/crt_tbl_prtntd_tbl.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+source conf/drillTestConfig.properties
+
+${DRILL_HOME}/bin/sqlline -n ${USERNAME} -p ${PASSWORD} -u "jdbc:drill:schema=dfs.Join;drillbit=${DRILL_STORAGE_PLUGIN_SERVER}"  --run=${DRILL_TEST_DATA_DIR}/Datasources/join/crt_tbls_partition_by_l.ddl
+${DRILL_HOME}/bin/sqlline -n ${USERNAME} -p ${PASSWORD} -u "jdbc:drill:schema=dfs.Join;drillbit=${DRILL_STORAGE_PLUGIN_SERVER}"  --run=${DRILL_TEST_DATA_DIR}/Datasources/join/crt_tbls_partition_by_r.ddl
+
+# it seems that sqlline does not exit, if one of the queries failed.
```
Comment (Agirish) [inline framework/resources/Datasources/join/crt_tbl_prtntd_tbl.sh]: The queries could fail for other reasons as well - not just absence of Sqlline. May be update the comment?

File: framework/resources/Datasources/join/crt_tbls_partition_by_r.ddl (hunk 1 above)
Comment (kfaraaz) [inline framework/resources/Datasources/join/crt_tbls_partition_by_r.ddl]: They are commented for now, as there are dependent tests that fail, due to these failing CTAS statements. Will be uncommented once fix is in.

File: framework/resources/Datasources/join/crt_tbls_partition_by_l.ddl [hunk 3]
```diff
@@ -0,0 +1,22 @@
+DROP TABLE IF EXISTS l_tblprtnby_intcl;
+CREATE TABLE l_tblprtnby_intcl PARTITION BY( col_int ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_l`;
+--DROP TABLE IF EXISTS l_tblprtnby_chrcl;
+--CREATE TABLE l_tblprtnby_chrcl PARTITION BY( col_chr ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_l`;
+--DROP TABLE IF EXISTS l_tblprtnby_vrchrcl1;
+--CREATE TABLE l_tblprtnby_vrchrcl1 PARTITION BY( col_vrchr1 ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_l`;
+--DROP TABLE IF EXISTS l_tblprtnby_vrchrcl2;
+--CREATE TABLE l_tblprtnby_vrchrcl2 PARTITION BY( col_vrchr2 ) AS SELECT * FROM dfs.`/drill/testdata/join/typeall_l`;
```
Comment (kfaraaz) [inline framework/resources/Datasources/join/crt_tbls_partition_by_l.ddl]: same reason. Due to an open CTAS bug.

**Comment by Agirish** [comment]
Can you squash these two commits into 1?

`git rebase -i HEAD~2` 
Replace 'pick' with 's' for the second commit.
Force push to your branch. 

The PR will get updated automatically.

File: framework/resources/Datasources/join/crt_tbl_prtntd_tbl.sh (hunk 2 above)
Comment (kfaraaz) [inline framework/resources/Datasources/join/crt_tbl_prtntd_tbl.sh]: I can update the bug in a later commit, as this is just a comment update.

File: framework/resources/Datasources/join/crt_tbls_partition_by_r.ddl (hunk 1 above)
Comment (Agirish) [inline framework/resources/Datasources/join/crt_tbls_partition_by_r.ddl]: Thanks for clarifying.

File: framework/resources/Datasources/join/crt_tbls_partition_by_l.ddl (hunk 3 above)
Comment (Agirish) [inline framework/resources/Datasources/join/crt_tbls_partition_by_l.ddl]: Thanks!

**Comment by kfaraaz** [comment]
Done, can you please check.

**Comment by Agirish** [comment]
+1

**Comment by kfaraaz** [comment]
thanks Abhishek,
